packages/core/src/actions/execute.ts

```diff
@@ -669,109 +932,353 @@ export const runEntireDeploymentProcess = async (
   const { status, failureAction, executionReceipts, batches } =
     await executeBatchActions(
       networkLeaves,
-      BigInt(parsedConfig.chainId),
       sphinxModule,
       BigInt(parsedConfig.blockGasLimit),
       humanReadableActions,
       executionMode,
-      signer,
-      provider,
       executeActions,
-      estimateGas
+      estimateGas,
+      deploymentContext
     )
 
-  spinner.succeed(`Executed deployment.`)
-  spinner.start(`Checking final deployment status...`)
+  spinner?.succeed(`Executed deployment.`)
+  spinner?.start(`Checking final deployment status...`)
 
   ethersReceipts.push(...executionReceipts)
-  const receipts = ethersReceipts.map(convertEthersTransactionReceipt)
 
-  if (status === MerkleRootStatus.FAILED) {
-    spinner.fail(`Deployment failed.`)
-    return { receipts, batches, finalStatus: status, failureAction }
-  } else if (status === MerkleRootStatus.CANCELED) {
-    spinner.fail(`Deployment cancelled by user.`)
-    return { receipts, batches, finalStatus: status }
-  } else if (status === MerkleRootStatus.COMPLETED) {
-    spinner.succeed(`Deployment succeeded.`)
-    return { receipts, batches, finalStatus: status }
-  } else {
-    throw new Error(`Unknown status: ${status}`)
+  return handleStatus(status, batches, ethersReceipts, failureAction, spinner)
+}
+
+/**
+ * @notice Sorts an array of hex strings in ascending order. This function mutates the array.
+ */
+export const sortSigners = (arr: Array<TreeSigner>): void => {
+  arr.sort((a, b) => {
+    const aBigInt = BigInt(a.signer)
+    const bBigInt = BigInt(b.signer)
+
+    if (aBigInt < bBigInt) {
+      return -1
+    } else if (aBigInt > bBigInt) {
+      return 1
+    } else {
+      return 0
+    }
+  })
+}
+
+/**
+ * This is the primary function used for executing deployments in the deploy command, propose command simulation,
+ * and website backend. There is some logic in this function that may not be strictly necessary in all situations.
+ *
+ * We chose to reuse it in all situations because it is easier to maintain a single interface and because we want
+ * to improve the chance that bugs will get caught during the initial proposal simulation step rather than occuring
+ * when a deployment is actually happening on the website.
+ *
+ * The allow for this function to be reused in all three situations this function accepts a `deploymentContext` object
+ * which implements all the logic that needs to be different in each of the three situations.
+ *
+ * We aim to *only* use the `deploymentContext` for logic which absolutely cannot be shared. For example, logic for
+ * writing information about a deployment to the website backend is implemented in the `deploymentContext`.
+ *
+ * See documentation on the `DeploymentContext` type at the top of this file for more information of the specific fields
+ * and justification for why the logic implemented in them cannot be shared.
+ */
+export const compileAndExecuteDeployment = async (
```

Comment:
I think we should add docs somewhere (not sure where) saying something like:

> The `ExecutionMode.Platform` is used by the website in this function, which means it can't contain any local network operations for this mode, such as assigning roles to an arbitrary address. However, we can perform local network operations via dependency injection, as long as the injected dependency isn't used by the website. This is the purpose of the `injectRoles` function.